### ISSUE TITLE
About Transfer Learning in Espnet2

### ISSUE THREAD

**Issue by miziworld** [espnet/espnet#4290]
Hello, I want to develop cross language transfer learning in espnet2. According to espnet's documentation, this doesn't seem too difficult. However, I have a few questions that I have to ask.

1. When using cross language, does simply setting the --ignore_init_mismatch "true" option solve the problem of matching input and output sizes?
2. In general, it is known that the accuracy improves when transfer learning is used compared to when transfer learning is not used. Did the accuracy improve when you actually test it?
3. In my opinion, performance is expected to vary greatly depending on the selection of the language for the pre-trained model and the model language for transfer learning. Since each language has different grammar, vocabulary, word order, etc., it is expected that languages ​​with similar grammar will perform better. I want to test this, can you recommend which of the asr pretrained models has the largest size and highest accuracy in English/Chinese/Japanese models?
4. In the transfer learning process, can you tell which selection was most efficient when selecting the parts to leave or exclude from the encoder or decoder?

Always thank you for developers of ESPNet..

**Comment by sw005320** [comment]
@DanBerrebbi, can you answer them for me?
Some of them are related to your experience.

**Comment by DanBerrebbi** [comment]
Yes @sw005320,
Hi @miziworld, thank you for your questions, here are my answers : 

1. You should know that the way it works is initializing the new model with the pre-trained weights. If there is mismatch between the size of new model and pre-trained one, setting the --ignore_init_mismatch "true" will solve all the problems by initializing with the default way you did set up (such as xavier uniform) and not with the pre-trained model weights for the components where the mismatch occurs. (You will be notified of it in the train.log file) 
2. It is known to improve the accuracy for some set-up (similar languages, similar domains ...). In those cases, indeed we observed significant improvement with this ESPnet transfer learning option. 
3. Can you clarify a bit your request? What is your target language/dataset? Also we have a lot of models trained on different datasets for English for instance, so not only language but also domain is important (read speech, conversational speech, gender of speakers ...). So I may be more capable to give you advice if you give me some more precisions.
4. I'd say that I usually keep all the components of the pre-trained model (just drop the ones with size mismatches through --ignore_init_mismatch "true" but do not manually remove some other parts). If you use a decoder, it may worth trying one model that keeps all the components, and one model that do not use pre-trained weights for the decoder.  

Thank you for using ESPnet !

**Comment by miziworld** [comment]
Thank you so much for your kind and detailed reply. And I would like to ask a few more questions about this.

The reason I want to use transfer learning is 1. To train a pre-trained model with a large structure to increase the accuracy in a situation where the dataset is insufficient. For example, to perform the ASR function, a general model is Can I use transfer learning if I want to train on syllables or words and only want to recognize 5 simple commands?
2. Because the pre-trained model has a large model size and complex structure, it is intended to be used as a lightweight method. In the case of transfer learning, it seems to be used similarly to fine-tuning. When using transfer learning, is it meaningful to reduce the size by deleting some blocks of the encoder or decoder in addition to the initial weight setting?
3. Do the structures of the pre-trained model and the student model have to be unified? (For example, transformer-transformer, rnn-rnn...)
